### ISSUE TITLE
Gitlab.delete_project raises exception

### ISSUE THREAD

**Issue by realsugar** [NARKOZ/gitlab#138]
When I call, for example, 

```
Gitlab.delete_project(42)
```

I get this:

```
/Users/...skipped.../gems/gitlab-3.4.0/lib/gitlab/request.rb:25:in `parse': Couldn't parse a response body (Gitlab::Error::Parsing)
	from /Users/...skipped.../gems/gitlab-3.4.0/lib/gitlab/request.rb:10:in `block in <class:Request>'
	from /Users/...skipped.../gems/httparty-0.13.5/lib/httparty/request.rb:309:in `call'
	from /Users/...skipped.../gems/httparty-0.13.5/lib/httparty/request.rb:309:in `parse_response'
	from /Users/...skipped.../gems/httparty-0.13.5/lib/httparty/request.rb:279:in `block in handle_response'
	from /Users/...skipped.../gems/httparty-0.13.5/lib/httparty/response.rb:23:in `call'
	from /Users/...skipped.../gems/httparty-0.13.5/lib/httparty/response.rb:23:in `parsed_response'
	from /Users/...skipped.../gems/gitlab-3.4.0/lib/gitlab/request.rb:77:in `validate'
	from /Users/...skipped.../gems/gitlab-3.4.0/lib/gitlab/request.rb:59:in `delete'
	from /Users/...skipped.../gems/gitlab-3.4.0/lib/gitlab/client/projects.rb:101:in `delete_project'
	from /Users/...skipped.../gems/gitlab-3.4.0/lib/gitlab.rb:22:in `method_missing'
	from client.rb:14:in `run'
	from client.rb:18:in `<main>'
```

Project is being deleted from Gitlab successfully, though.

*TIP*
Seems like *parse* method that raises exception, can not handle empty response body.

If I delete a project using *curl*, I get this:

```
curl -i -v -X DELETE --header "PRIVATE-TOKEN: MY-TOKEN" "http://example.com/api/v3/projects/42"

> DELETE /api/v3/projects/42 HTTP/1.1
> User-Agent: curl/7.37.1
> Host: example.com
> Accept: */*
> PRIVATE-TOKEN: ...
>
< HTTP/1.1 200 OK
HTTP/1.1 200 OK
* Server nginx is not blacklisted
< Server: nginx
Server: nginx
< Date: Tue, 07 Jul 2015 08:32:26 GMT
Date: Tue, 07 Jul 2015 08:32:26 GMT
< Content-Type: application/json
Content-Type: application/json
< Content-Length: 4
Content-Length: 4
< Connection: keep-alive
Connection: keep-alive
< Status: 200 OK
Status: 200 OK
< ETag: "..."
ETag: "..."
< Cache-Control: max-age=0, private, must-revalidate
Cache-Control: max-age=0, private, must-revalidate
< X-Request-Id: dbd40ece-066f-4d9d-aeff-6378b5a09f94
X-Request-Id: dbd40ece-066f-4d9d-aeff-6378b5a09f94
< X-Runtime: 0.119886
X-Runtime: 0.119886

<
```

Last blank line of response means empty HTTP Response body I guess.

Thanks in advance for fixing this!

Regards,
Anton Sakharov

**Comment by NARKOZ** [comment]
When you delete a record the record is returned in response body. If you try to delete the record that doesn't exist you'll get an empty response because the record was already deleted. In that case gitlab client should return `false`.

**Comment by asedge** [comment]
I think this is the same issue that MR #137 is trying to fix.  The Gitlab API calls `::Projects::DestroyService.new(user_project, current_user, {}).execute` ([here](https://github.com/gitlabhq/gitlabhq/blob/master/lib/api/projects.rb#L245-L254)) and `execute` returns true if it was successful ([here](https://github.com/gitlabhq/gitlabhq/blob/master/app/services/projects/destroy_service.rb#L33)).  It would be nice if the Gitlab API returned a successful message rather than just `true`.

**Comment by NARKOZ** [comment]
@asedge Thanks. Gitlab API is inconsistent and that isn't specified in docs.

So, when a record gets deleted a response can be `true`, `false`, the resource as a JSON, or an empty response.

**Comment by asedge** [comment]
I believe this was fixed by #137.  Please re-open if that's not the case, thanks!
